controllers/multiclusterobservability/multiclusterobservability_controller.go

```diff
@@ -127,11 +131,38 @@ func (r *MultiClusterObservabilityReconciler) Reconcile(ctx context.Context, req
 		return ctrl.Result{}, err
 	}
 
-	//read image manifest configmap to be used to replace the image for each component.
-	if _, err = config.ReadImageManifestConfigMap(r.Client); err != nil {
+	mchCrdExists, err := util.CheckCRDExist(r.CrdClient, config.MCHCrdName)
+	if err != nil {
 		return ctrl.Result{}, err
 	}
 
+	if mchCrdExists {
+		mchList := &mchv1.MultiClusterHubList{}
+		podNamespace, found := os.LookupEnv("POD_NAMESPACE")
+		if !found {
+			podNamespace = config.GetDefaultMCONamespace()
+		}
+		mchistOpts := []client.ListOption{
+			client.InNamespace(podNamespace),
+		}
+		err := r.Client.List(context.TODO(), mchList, mchistOpts...)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// normally there should only one MCH CR in the cluster
+		if len(mchList.Items) == 1 {
+			mch := mchList.Items[0]
+			mchVer := mch.Status.CurrentVersion
+			if mchVer != "" && (mch.Status.CurrentVersion == mch.Status.DesiredVersion) {
+				//read image manifest configmap to be used to replace the image for each component.
+				if _, err = config.ReadImageManifestConfigMap(r.Client, mchVer); err != nil {
+					return ctrl.Result{}, err
+				}
+			}
```

Comment:
if mch.Status.CurrentVersion != mch.Status.DesiredVersion, just return.

Reply:
I think we should just skip reading the imahe manifests if mch.Status.CurrentVersion != mch.Status.DesiredVersion. the other steps should still be processed.